src/components/MovieCard.vue

```diff
@@ -0,0 +1,79 @@
+<script setup lang="ts">
+export interface Movie {
+  title: string
+  poster: string
+  genre: string
+  releaseDate: number
+}
+
+const props = defineProps({
+  movie: {
+    type: Object as () => Movie,
+    required: true
+  }
+})
```

Comment:
You can use generic for props typization
https://vuejs.org/guide/typescript/composition-api